### DISCUSSION TITLE
one2many, how to query the last one in many collection ?

### DISCUSSION THREAD

**Discussion by NMTuan** [directus/directus#16047] [Archive]
i have 2 collection: 
news: id, views
view_logs: id, date_created, news_id

view_logs.news_id --> mary 2 one --> news
news.views --> one 2 mary --> view_logs.nes_id

now, i want created page to show news list, 
show the last view.date_created, and use the last view.date_created sort.

how to do it with js_sdk.

```js
directus.items('news').readByQuery({
    meta: '*',
    fields: [
        'id',
        'count(views)',
        'views.*???'
    ],
    sort: ['???']
})
```

**Comment by br41nslug** [comment]
I believe you cannot sort by related values so you may have to turn this query around.
Something along these lines should do it:
```js
directus.items('view_logs').readByQuery({
    fields: [ 'id', 'news_id.*', 'date_created' ],
    groupBy: ['news_id'],
    aggregate: { 'max': 'date_created' },
    sort: ['-date_created'],
})
```
https://docs.directus.io/reference/query.html#sort
https://docs.directus.io/reference/query.html#aggregation-grouping

**Reply by nickrum** [reply]
Sorting by a related value should be possible since #12084 🤔

**Reply by br41nslug** [reply]
Ah outdated knowledge... However in the comments of that PR it seems that won't work for the o2m relation here.

**Reply by licitdev** [reply]
Support for nested o2m sorting will be added via https://github.com/directus/directus/pull/14798.
Additionally, when using aggregate or group, the fields will be ignored.

https://github.com/directus/directus/blob/13f4ec027d01f0b708dd62b6adc8a7f84f897b7b/api/src/utils/get-ast-from-query.ts#L53-L67

**Reply by NMTuan** [reply]
Will the query be slow when the log collection is too large?

or create a new field:  news.date_updated. whenever a log is inserted, update it ?

**Reply by licitdev** [reply]
I believe a large log collection should not affect the query performance of other collections.
In addition, there is control over activity and revision tracking per collection.
Ref: https://docs.directus.io/configuration/data-model/collections.html#accountability
